docs/source/learn/core_notebooks/Gaussian_Processes.rst

```diff
@@ -0,0 +1,267 @@
+      K_1       &  K_1^{*}   & K_2       & K_2^{*}   & K_1 + K_2  & K_1^{*} + K_2^{*}  \\
+      K_1^{*^T} & K_1^{**} & K_2^{*^T} & K_2^{**} & K_1^{*^T}+K_2^{*^T} & K_1^{**}+K_2^{**}
+    \end{bmatrix}
+  \right) \,.
+
+Using the joint distribution to obtain the conditional distribution of :math:`f_1^*`
+with the contribution due to :math:`f_1 + f_2` factored out, we get
+
+.. math::
+  f_1^* \mid f_1 + f_2 \sim \text{N}\left(
+    m_1^* + K_1^{*^T}(K_1 + K_2)^{-1}\left[f_1 + f_2 - m_1 - m_2\right] \,,\,
+    K_1^{**} - K_1^{*^T}(K_1 + K_2)^{-1}K_1^* \right) \,.
+
+
+These equations show how to break down GP models into individual components to see how each
+contributes to the data.  For more information, check out `David Duvenaud's PhD
+thesis <https://www.cs.toronto.edu/~duvenaud/thesis.pdf>`_.
+
+The GP objects in PyMC keeps track of these marginals automatically.  The
+following code sketch shows how to define the conditional distribution of
+:math:`f_2^*`.  We use `gp.Marginal` in the example, but the same works for
+other implementations.  The first block fits the GP prior.  We denote
+:math:`f_1 + f_2` as just :math:`f` for brevity::
+
+    with pm.Model() as model:
+        gp1 = pm.gp.Marginal(mean_func1, cov_func1)
+        gp2 = pm.gp.Marginal(mean_func2, cov_func2)
+
+        # gp represents f1 + f2.
+        gp = gp1 + gp2
+
+        f = gp.marginal_likelihood("f", X, y, sigma)
+
+        idata = pm.sample(1000)
+
+
+To construct the conditional distribution of :code:`gp1` or :code:`gp2`, we
+also need to include the additional arguments, :code:`X`, :code:`y`, and
+:code:`sigma`::
+
+    with model:
+        # conditional distributions of f1 and f2
+        f1_star = gp1.conditional("f1_star", X_star,
+                                  given={"X": X, "y": y, "sigma": sigma, "gp": gp})
+        f2_star = gp2.conditional("f2_star", X_star,
+                                  given={"X": X, "y": y, "sigma": sigma, "gp": gp})
+
+        # conditional of f1 + f2, `given` not required
+        f_star = gp.conditional("f_star", X_star)
+
+This second block produces the conditional distributions.  Notice that extra
+arguments are required for conditionals of :math:`f1` and :math:`f2`, but not
+:math:`f`.  This is because those arguments are cached when
+:code:`.marginal_likelihood` is called on :code:`gp`.
+
+.. note::
+  When constructing conditionals, the additional arguments :code:`X`, :code:`y`,
+  :code:`sigma` and :code:`gp` must be provided as a dict called `given`!
+
+Since the marginal likelihoood method of :code:`gp1` or :code:`gp2` weren't called,
+their conditionals need to be provided with the required inputs.  In the same
+fashion as the prior, :code:`f_star`, :code:`f1_star` and :code:`f2_star` are random
+variables that can now be used like any other random variable in PyMC.
+
+Check the notebooks for detailed demonstrations of the usage of GP functionality
+in PyMC.
```

Comment:
Also missing link, maybe to the gaussian process section of the example gallery? 

https://www.pymc.io/projects/examples/en/latest/gallery.html#gaussian-processes

Reply:
I'm having some trouble with this one. Do you know how to write a reference to a section in the gallery rather than a specific notebook? I tried two approaches.

First, I tried using a static link, like:

    Check the `notebooks <https://www.pymc.io/projects/examples/en/latest/gallery.html#gaussian-processes>`_ for detailed demonstrations of the usage of GP functionality in PyMC.

But pre-commit would fail with:

    Check no links that should be cross-references are in the docs.......................Failed
     - hook id: no-internal-links
     - exit code: 1

Then, I tried doing a reference with:

    Check the \`notebooks <gaussian-processes>`_ for detailed demonstrations of the usage of GP functionality
    in PyMC.

but it would generate a broken link. It expect the reference to be somewhere in /learn/core_notebooks/ folder rather than looking in the examples gallery.

Reply:
I opened https://github.com/pymc-devs/pymc-examples/pull/546 to be able to do this. After it is merged you'll be able to reference sections with their folder name (or what is the same, the keys in [this dict](https://github.com/pymc-devs/pymc-examples/blob/0c39a19c4c91dcf1a3b57222eb61677979f04405/sphinxext/thumbnail_extractor.py#L96-L109))

Reply:
okay that did the trick! The reference works properly now. Thanks for your help.